packages/block-editor/src/components/global-styles/use-global-styles-output.js

```diff
@@ -544,10 +538,20 @@ export function getLayoutStyles( {
 		);
 		// For backwards compatibility, ensure the legacy block gap CSS variable is still available.
 		if ( selector === ROOT_BLOCK_SELECTOR && hasBlockGapSupport ) {
-			ruleset += `${ ROOT_CSS_PROPERTIES_SELECTOR } { --wp--style--block-gap: ${ gapValue }; }`;
+			rootStyles += `--wp--style--block-gap: ${ gapValue };`;
 		}
 	}
 
+	if ( ROOT_BLOCK_SELECTOR === selector && useRootPadding ) {
```

Comment:
I think I need to move padding CSS property generation back to where it was.

Consider https://github.com/WordPress/gutenberg/blob/1d99f4c4b1abc92359b7f2fe3246ba6f9a8bb313/packages/edit-post/src/components/layout/index.js#L113-L113

Here `getLayoutStyles` generates base layout styles - it's in `edit-post` package.

Without access to global styles in the post editor, `getLayoutStyles` cannot generate the root padding properties.